packages/devextreme/js/__internal/grids/new/grid_core/columns_controller/utils.ts

```diff
@@ -157,3 +174,32 @@ export function getColumnByIndexOrName(
 
   return column;
 }
+
+export const getValueDataType = function (value: unknown): DataType {
+  let dataType: string | undefined = type(value);
+  if (dataType !== 'string' && dataType !== 'boolean' && dataType !== 'number' && dataType !== 'date' && dataType !== 'object') {
+    dataType = undefined;
+  }
+
+  return dataType as DataType;
+};
+
+export const getColumnFormat = function (column: Partial<Pick<Column, 'format' | 'dataType'>>): Format | undefined {
+  if (column.format) {
+    return column.format;
+  }
+
+  if (column.dataType === 'date' || column.dataType === 'datetime') {
+    return 'shortDate';
+  }
+
+  return undefined;
+};
+
+export function generateColumns(
+  firstItem: DataObject | Record<string, unknown>,
```

Comment:
The ```DataObject``` type is already ```Record<string | unknown>```
But the ```firstItem``` may not exist, so let's change it the following way:

```suggestion
  firstItem: DataObject | null,
```